climada/entity/impact_funcs/test/test_base.py

```diff
@@ -60,6 +59,34 @@ def test_from_sigmoid(self):
         self.assertEqual(imp_fun.haz_type, 'RF')
         self.assertEqual(imp_fun.id, 2)
 
+    def test_from_poly_s_shape(self):
+        """Check default impact function: polynomial s-shape"""
+        inten = (0, 5, 1)
+        impf = ImpactFunc.from_poly_s_shape(
+            inten, threshold=0.2, half_point=1, upper_limit=0.8, exponent=4,
+            haz_type='RF', impf_id=2, intensity_unit='m'
+            )
+        correct_mdd = np.array([0, 0.4, 0.76995746, 0.79470418, 0.79843158])
```

Comment:
It's not clear why these are the correct values. You should check that the function is
* zero below the threshold
* zero at the threshold
* 0.5x the upper limit at the half_point *(this is covered)*
* (very) close to the upper limit somewhere 10x half_point
* behaving as expected when threshold < 0, half_point < threshold, upper_limit < 0, exponent < 0.

Reply:
In this case, I opted for a direct check of the values as one can easily recompute by hand what the correct values are. I think the checks you propose are less good as they do not directly test the mathematical formula. I can add some tests for the edge cases, although I think this is a clear overkill here.

Reply:
Please don't patronize the readers of your code ("can easily be computed"). Can't you find one or multiple sets of parameters where the true values can be calculated *without* a calculator?

> I can construct a function that fulfils all your tests but that does not represent the correct formula.

That is true for most tests. I agree that some tests have to be specific, but the corner cases have to be checked nonetheless.